### PR TITLE
handle errors through FsError and FS saga cleanup

### PR DESCRIPTION
**Pull request by songgao**
With this, we should be showing the red banner from https://github.com/keybase/client/pull/13315 for FS errors most of time.

Also while testing this, I'm noticing the upload banner is still kind of flaky, especially for small files. I'll try to fix it when I get to KBFS-3325.

### CODE REVIEW

File: shared/reducers/fs.js [hunk 1]
```diff
@@ -204,11 +203,37 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
...
-    case FsGen.editFailed:
-      // $FlowFixMe
-      return state.setIn(['edits', action.payload.editID, 'status'], 'failed')
     case FsGen.fsError:
-      return state.setIn(['errors', Constants.makeUUID()], action.payload.error)
+      const {erroredAction, error} = action.payload.error
+      logger.error('error (fs)', erroredAction.type, error)
+      const nextState: Types.State = state.setIn(['errors', Constants.makeUUID()], action.payload.error)
+
+      switch (erroredAction.type) {
```
Comment (songgao) [inline shared/reducers/fs.js]: @chrisnojima I ended up moving the error stuff for download, upload, and edits here, so saga don't need to worry about what component need information about errors. It also don't require dispatching multiple actions from a single saga action handler. Please let me know if you think there's a better way to do this.

File: shared/actions/fs/index.js [hunk 2]
```diff
@@ -354,7 +355,8 @@ const getWaitDuration = (endEstimate: ?number, lower: number, upper: number): nu
 }
 
 let polling = false
-function* pollSyncStatusUntilDone(): Saga.SagaGenerator<any, any> {
+function* pollSyncStatusUntilDone(action: FsGen.FsActivityPayload): Saga.SagaGenerator<any, any> {
+  console.log('pollllllll')
```
Comment (jzila) [inline shared/actions/fs/index.js]: :fire:

File: shared/actions/fs/index.js [hunk 3]
```diff
@@ -516,13 +523,18 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
...
+function* loadMimeType(action: FsGen.MimeTypeLoadPayload) {
+  try {
+    yield Saga.call(_loadMimeType, action.payload.path, action.payload.refreshTag)
+  } catch (error) {
+    yield Saga.put(makeUnretriableErrorHandler(action)(error))
+  }
+}
 
-const commitEdit = (action: FsGen.CommitEditPayload, state: TypedState) => {
+const commitEdit = (state: TypedState, action: FsGen.CommitEditPayload) => {
```
Comment (jzila) [inline shared/actions/fs/index.js]: Since we always return a promise now, you can type the return type here.

Comment (chrisnojima) [reply shared/actions/fs/index.js]: again, not really needed. flow will tell you when it needs to be typed

File: shared/actions/fs/index.js (hunk 2 above)
Comment (songgao) [reply shared/actions/fs/index.js]: Ah; I only realized in my follow-up PR, so removed it there: https://github.com/keybase/client/pull/13391

Comment (songgao) [reply shared/actions/fs/index.js]: Actually I'll remove it here so it's never in master. Thanks

File: shared/actions/fs/index.js (hunk 3 above)
Comment (songgao) [reply shared/actions/fs/index.js]: This is used in `actionToPromise` below, so it's not vague to humans. @jzila I'm gonna keep it this way unless you have a strong opinion here.

**Comment by songgao** [comment]
Bump 🕺🏿

**Pull request merged**